feat(KaizenProvider): add KaizenProvider to @kaizen/components (#3239)
cultureamp/kaizen-design-system: packages/components/src/KaizenProvider/docs/KaizenProvider.stories.mdx

import { Meta } from '@storybook/addon-docs'
import { CATEGORIES } from "../../../../../storybook/constants"

> <Meta title={`${CATEGORIES.components}/KaizenProvider`} />

Reviewer: I honestly don't like the `components` category. It's too broad and not useful. Categories like Graphics/Form/Layout are better (separate discussion from this PR though). Not that I know what to group this under though 😂

Author: I'd group it under `Global` -- But yeah a wider discussion on this, I know the designers have already come up with groupings

# KaizenProvider

> Provides the necessary theming context for all components, as well as facilitate the use of global UI components and translations.

Reviewer 2: ```suggestion
Provides the necessary theming context for all components, as well as facilitates the use of global UI components and translations.
```

## Usage

> Ideally, the `KaizenProvider` should wrap your entire application, and that there only be one per application.

Reviewer 2: ```suggestion
Ideally, the `KaizenProvider` should wrap your entire application, and that there only be one per application.
```
That last part of the sentence needs a rewrite.

```jsx
import { KaizenProvider } from '@kaizen/components';

<KaizenProvider>
  <App />
</KaizenProvider>
```

## Customising the theme

Out of the box, the `KaizenProvider` utilises the `defaultTheme` from `@kaizen/design-tokens`. However if you want to roll your own theme, we provide a `ThemeManager` function which can apply your custom theme.

```jsx
import { ThemeManager } from '@kaizen/components'

export const themeManager = new ThemeManager({
  themeKey: "custom",
  spacing: {
    md: "1rem",
  },
});

<KaizenProvider themeManager={themeManager}>
  <App />
</KaizenProvider>

> ```

Author: Not quite sure where this belongs any more, on one hand it's the API for `KaizenProvider` on the other, it's only used for theme-ing and shouldn't be used in most cases?

Reviewer: You mean the docs? Once you have i18n in the KaizenProvider, you could split it to be `KaizenProvider/Theming` and `KaizenProvider/Internationalisation`. Or you could even have the subcategory now if you'd like.

## Using the theme

> For more details on how theme-ing works, checkout our [design tokens documentation](/story/design-tokens-how-to-use-design-tokens-in-react-components--page).

Reviewer: ```suggestion
For more details on how theming works, checkout our [design tokens documentation](/story/design-tokens-how-to-use-design-tokens-in-react-components--page).
```

Reviewer: But...English 😂
